### ISSUE TITLE
unable to create multiple sessions after updating Android

### ISSUE THREAD

**Issue by mhuesch** [t184256/nix-on-droid#168]
I recently installed an Android system update, which I believe took me from Android 11 to 12 (I'm on 12 now, and I can't check backwards what I was on before, but believe it was a major-version lower).

before, everything worked perfectly.

now, when I hit the `➕ New session` option after long-pressing the app icon, I see:

```
Failed to read the app data. Cannot start the app
```

and I remain with only one nix-on-droid session.

happy to look for other debug info that might inform what's going on. I didn't see any other useful info printed out.

**Comment by Gerschtli** [comment]
Could you check, if it works in [termux](https://termux.com/)? Maybe we need to update the nix-on-droid-app..

**Comment by mhuesch** [comment]
@Gerschtli good idea - I just checked Termux (version 0.118.0) and it is able to successfully create multiple sessions.

**Comment by Gerschtli** [comment]
> This branch is 24 commits ahead, 739 commits behind termux:master.

@t184256 Could you rebase [nix-on-droid-app](https://github.com/t184256/nix-on-droid-app) to latest master and test if it still works? Maybe would be a good idea to rebase every time we release a new version.

I would do it by myself but I don't know how to build/test it :D

**Comment by t184256** [comment]
That's no small request, unfortunately, doubt I'll have the time to do it any reasonably soon.

While Nix-specific modifications are rather thin (app prefix change, a function with some extra symlink resolving magic, change of a download url), I've picked a rather unorthodox base for it: a branch where the paid addons have been integrated (styling, etc) into the main apk.

So even if I find a weekend to reapply my patches on top of the latest patches and create a test APK so that the reporter checks whether this specific thing is fixed, I guess nobody's gonna redo the addon merging.

We need an "Android part maintainer needed" banner or a pinned issue or something =(

**Comment by t184256** [comment]
No, I haven't. Are there any bounty-meets-our-saviour systems that would function without any involvement from my side? In that case I could reopen issue tracker for the Java app and move some bugs there.

**Comment by t184256** [comment]
Sigh. I've tried an S8 Tab Ultra with Android 12 and it even more broken there, "can't turn BTI protection". Sessions can be created from a sidebar, but not from an app icon.
